packages/nats/test/NATSIntegration.test.ts

```diff
@@ -0,0 +1,89 @@
+          })
+
+          expect(connectionLayer).toBeDefined()
+          expect(publisherLayer).toBeDefined()
+          expect(subscriberLayer).toBeDefined()
+
+          // Test message handling with mock data
+          const mockMsg: Msg = {
+            data: new TextEncoder().encode("integration test message"),
+            string: () => "integration test message",
+            json: () => ({ test: "data", id: 123 }),
+            subject: "test.subject",
+            headers: { "Content-Type": "application/json" },
+            reply: "test.reply"
+          } satisfies Partial<Msg> as Msg
+
+          const natsMessage = NATSMessage.make(mockMsg)
+
+          // Verify message utilities work correctly
+          expect(NATSMessage.data(natsMessage)).toEqual(
+            new TextEncoder().encode("integration test message")
+          )
+          expect(NATSMessage.string(natsMessage)).toBe("integration test message")
+          expect(NATSMessage.json(natsMessage)).toEqual({ test: "data", id: 123 })
+          expect(NATSMessage.subject(natsMessage)).toBe("test.subject")
+          expect(NATSMessage.headers(natsMessage)).toEqual({ "Content-Type": "application/json" })
+          expect(NATSMessage.reply(natsMessage)).toBe("test.reply")
+
+          // Test publish message structure
+          const publishMessage: NATSMessage.NATSPublishMessage = {
+            subject: "test.publish",
+            data: new TextEncoder().encode("publish test"),
+            options: {
+              reply: "test.reply",
+              headers: { "X-Custom": "header" }
+            }
+          }
+
+          expect(publishMessage.subject).toBe("test.publish")
+          expect(publishMessage.data).toEqual(new TextEncoder().encode("publish test"))
+          expect(publishMessage.options?.reply).toBe("test.reply")
+          expect(publishMessage.options?.headers).toEqual({ "X-Custom": "header" })
+
+          // Verify TypeId uniqueness and context tags
+          expect(NATSConnection.TypeId).toBeDefined()
+          expect(NATSMessage.TypeId).toBeDefined()
+          expect(NATSPublisher.TypeId).toBeDefined()
+          expect(NATSSubscriber.TypeId).toBeDefined()
+
+          expect(NATSConnection.NATSConnection).toBeDefined()
+          expect(NATSMessage.NATSMessage).toBeDefined()
+          expect(NATSPublisher.NATSPublisher).toBeDefined()
+          expect(NATSSubscriber.NATSSubscriber).toBeDefined()
+
+          // Test that different TypeIds are unique
+          expect(NATSConnection.TypeId).not.toBe(NATSMessage.TypeId)
+          expect(NATSMessage.TypeId).not.toBe(NATSPublisher.TypeId)
+          expect(NATSPublisher.TypeId).not.toBe(NATSSubscriber.TypeId)
+
+          // Verify that functions for creating instances exist
+          expect(NATSConnection.makeConnection).toBeDefined()
+          expect(NATSPublisher.make).toBeDefined()
+          expect(NATSSubscriber.make).toBeDefined()
+        }).pipe(TestServices.provideLive),
```

Comment:
can you write a test that is actually usesful and that tests a publish and subscribe for real ?